integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanChargePaymentWithAdvancedPaymentAllocationTest.java

```diff
@@ -0,0 +1,325 @@
+    private static final String LP_REPAYMENTS = "1";
+    private static final String LP_REPAYMENT_PERIOD = "1";
+    private static final String EXPECTED_DISBURSAL_DATE = "04 March 2011";
+    private static final String LOAN_APPLICATION_SUBMISSION_DATE = "03 March 2011";
+    private static final String LOAN_TERM_FREQUENCY = "1";
+    private static final String INDIVIDUAL_LOAN = "individual";
+    private static final BusinessDateHelper BUSINESS_DATE_HELPER = new BusinessDateHelper();
+    private static final DateTimeFormatter DATE_FORMATTER = new DateTimeFormatterBuilder().appendPattern(DATETIME_PATTERN).toFormatter();
+    private static RequestSpecification requestSpec;
+    private static ResponseSpecification responseSpec;
+    private static LoanTransactionHelper loanTransactionHelper;
+    private static AccountHelper accountHelper;
+    private static PeriodicAccrualAccountingHelper periodicAccrualAccountingHelper;
+    private static Integer commonLoanProductId;
+    private static PostClientsResponse client;
+    private static BusinessDateHelper businessDateHelper;
+    private static SchedulerJobHelper scheduleJobHelper;
+    private static SavingsAccountHelper savingsAccountHelper;
+    private static SavingsProductHelper savingsProductHelper;
+
+    @BeforeAll
+    public static void setup() {
+        Utils.initializeRESTAssured();
+        ClientHelper clientHelper = new ClientHelper(requestSpec, responseSpec);
+        requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        requestSpec.header("Fineract-Platform-TenantId", "default");
+        responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+
+        loanTransactionHelper = new LoanTransactionHelper(requestSpec, responseSpec);
+        accountHelper = new AccountHelper(requestSpec, responseSpec);
+        periodicAccrualAccountingHelper = new PeriodicAccrualAccountingHelper(requestSpec, responseSpec);
+        final Account assetAccount = accountHelper.createAssetAccount();
+        final Account incomeAccount = accountHelper.createIncomeAccount();
+        final Account expenseAccount = accountHelper.createExpenseAccount();
+        final Account overpaymentAccount = accountHelper.createLiabilityAccount();
+        client = clientHelper.createClient(ClientHelper.defaultClientCreationRequest());
+        businessDateHelper = new BusinessDateHelper();
+        scheduleJobHelper = new SchedulerJobHelper(requestSpec);
+        savingsAccountHelper = new SavingsAccountHelper(requestSpec, responseSpec);
+        savingsProductHelper = new SavingsProductHelper();
+        commonLoanProductId = createLoanProduct("500", "15", "4", assetAccount, incomeAccount, expenseAccount, overpaymentAccount);
+    }
+
+    @Test
+    public void chargePaymentWithDefaultAllocationRuleTest() {
```

Comment:
Please add some extra test cases:
- What if the loan transaction amount is lower than the charge outstanding amount
- What if the loan transaction amount is greater than the charge outstanding amount
- What if there are allocation rules for CHARGE_PAYMENT transaction

Reply:
job picks up the charges need to be paid, and creates the transaction with the exact amount of the charge.
Do ou have any idea how can I test it with different transaction amounts?